src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
(For future readers) We discussed on appear.in, and decided to:

1. mimic the existing functionality in the existing load_class: https://github.com/jupyter/notebook/blob/a8f6265d07378d7e96d63cb77c2641de52cd3a4a/notebook/static/base/js/utils.js#L701
2. Propose an messaging spec change to make the comm target_module mandatory, deprecate the existing registry.
3. Provide a way to override system import paths to override things/load things dynamically.